include/irr/core/math/aabb2dSIMD.h

```diff
@@ -8,90 +8,92 @@
 #include <iostream>
 
 #define SIGN_FLIP_MASK_XY	core::vector3du32_SIMD(0x80000000, 0x80000000, 0x00000000, 0x00000000)
-#define SIGN_FLIP_MASK_XYZW core::vector3du32_SIMD(0x80000000, 0x80000000, 0x80000000, 0x80000000)
-
 
 namespace irr { namespace core {
 
+
 class aabb2dSIMDf
 {
 public:
 	inline aabb2dSIMDf()
-		: box( FLT_MAX, FLT_MAX, FLT_MAX, FLT_MAX ) {};
+		: internalBoxRepresentation(-FLT_MAX, -FLT_MAX, -FLT_MAX, -FLT_MAX) {};
 
-	inline aabb2dSIMDf(const core::vectorSIMDf& _box)
-		: box(_box ^ SIGN_FLIP_MASK_XY) {};
+	inline explicit aabb2dSIMDf(const core::vectorSIMDf& _box)
+		: internalBoxRepresentation(_box^ SIGN_FLIP_MASK_XY) {};
 
-	inline void addBox(const aabb2dSIMDf& _box)		
-	{ 
-		box = core::max_(box, _box.getBounds()); 
+	inline void addBox(const aabb2dSIMDf& _box)
+	{
+		internalBoxRepresentation = core::max_(internalBoxRepresentation, _box.getBounds());
 	}
 
-	inline void addPoint(const core::vectorSIMDf& point)	
+	inline void addPoint(const core::vectorSIMDf& point)
 	{
-		addBox(point); 
+		addBox(aabb2dSIMDf(point));
 	}
 
-	inline core::vectorSIMDf getBounds() const
+	inline core::vectorSIMDf getMinEdge() const
 	{
-		return box ^ SIGN_FLIP_MASK_XY;
+		core::vectorSIMDf minEdge = getBounds();
+		minEdge.makeSafe2D();
+		return minEdge;
 	}
 
-
-	//! Resets the bounding box to a one-point box.
-	/** \param x X coord of the point.
-	\param y Y coord of the point. */
-	inline void reset(float x, float y)
+	inline core::vectorSIMDf getMaxEdge() const
 	{
-		box = core::vectorSIMDf(-x, -y, x, y);
+		core::vectorSIMDf maxEdge = internalBoxRepresentation.zwzw();
+		maxEdge.makeSafe2D();
+		return maxEdge;
 	}
 
-	//! Resets the bounding box.
-	/** \param initValue New box to set this one to. */
-	inline void reset(const aabb2dSIMDf& initValue)
+	inline core::vectorSIMDf getBounds() const
 	{
-		box = initValue.box;
+		return internalBoxRepresentation ^ SIGN_FLIP_MASK_XY;
 	}
 
 	//! Resets the bounding box to a one-point box.
-	/** \param initValue New point. */
-	//void reset(const aabb2dSIMDf& initValue);
+	/** \param x X coord of the point.
+	\param y Y coord of the point. */
+	inline void reset(const core::vectorSIMDf& point)
```

Comment:
this could actually be a template function of the base class
```cpp
inline void reset(const T& point)
{
   CRTP<T>::operator=(CRTP<T>(point));
}
```

Reply:
it can be shared between aabbSIMDi and aabbSIMDu only, aabbSIMDf needs also sign flip operation

Reply:
Wrong.

aabb2dSIMDf will perform sign flip when calling a constructor from a single point